tests/php/validation/test-class-amp-validation-manager.php

```diff
@@ -1950,11 +1902,16 @@ public function test_decorate_shortcode_and_filter_source() {
 					'name'     => 'wp-includes',
 					'function' => 'prepend_attachment',
 				],
-				[
-					'type'     => 'core',
-					'name'     => 'wp-includes',
-					'function' => 'wp_make_content_images_responsive',
-				],
+			];
+		}
+
+		// This will be called after `do_shortcode` in WP 6.4 and later.
+		// @see <https://core.trac.wordpress.org/ticket/58853>.
+		if ( version_compare( get_bloginfo( 'version' ), '6.4-alpha', '<' ) ) {
```

Comment:
When doing `version_compare()`, I have found it useful to strip out the part after the hyphen as it can mess up the comparison.

```suggestion
		if ( version_compare( strtok( get_bloginfo( 'version' ), '-' ), '6.4', '<' ) ) {
```